### ISSUE TITLE
bug: redundant_env_clone lint misses UFCS-style Env::clone(&env) calls

### ISSUE THREAD

**Issue by mallison031** [Tollcraft/soroban-cost-linter#76]
## Description

`RedundantEnvClone` only matches `hir::ExprKind::MethodCall` where the method name is `clone`:

[`soroban_cost_lints/src/lib.rs`](https://github.com/Tollcraft/soroban-cost-linter/blob/main/soroban_cost_lints/src/lib.rs)
```rust
if let hir::ExprKind::MethodCall(path_segment, receiver, _args, _span) = expr.kind
    && path_segment.ident.name.as_str() == "clone"
```

Code that clones an `Env` via the fully-qualified/UFCS form — `Env::clone(&env)` or `Clone::clone(&env)` — takes the `hir::ExprKind::Call` path instead of `MethodCall`, so it never reaches this check. The exact anti-pattern the lint exists to catch passes through silently whenever it's written this way, which is common in generic or macro-expanded code.

## What "done" looks like
* `.clone()` on an `Env` receiver is detected regardless of whether it's written as a method call or in fully-qualified/UFCS form.
* A new fixture case using `Env::clone(&env)` (or `Clone::clone(&env)`) is confirmed to trigger the lint.
* The existing method-call-syntax detection and the `#[allow(...)]` suppression mechanism continue to pass unchanged.

## Implementation guidelines
* Key file: [`soroban_cost_lints/src/lib.rs`](https://github.com/Tollcraft/soroban-cost-linter/blob/main/soroban_cost_lints/src/lib.rs) (`RedundantEnvClone::check_expr`).
* Extend the match to also handle `hir::ExprKind::Call` where the callee path resolves to `Clone::clone` (or `Env::clone`) with an `Env`-typed argument.
* Add the new fixture case to [`soroban_cost_lints/ui/main.rs`](https://github.com/Tollcraft/soroban-cost-linter/blob/main/soroban_cost_lints/ui/main.rs) alongside the existing `bad_clone_env` case.

## PR guidelines
* Get assigned before starting.
* PR description must include: `Closes #[this issue]`.

---

### **Contact & Support**
- [Telegram](https://t.me/+Gflo5jZStw1jMjE0)
- [Discord](https://discord.gg/5aprtMSyR)

---
### 📋 Before you start
Please read our [Code Quality Standards](https://github.com/Tollcraft/soroban-cost-linter/blob/main/CONTRIBUTING.md#code-quality-standards) in [`CONTRIBUTING.md`](https://github.com/Tollcraft/soroban-cost-linter/blob/main/CONTRIBUTING.md). Before submitting a PR, ensure you run:
- cargo fmt --all -- --check
- cargo clippy --workspace --all-targets -- -D warnings
- cargo test --workspace

**Comment by mikevill20** [comment]
@mikevill20 has applied to work on this issue as part of the **Stellar Wave Program**'s 7th wave.



> Hello maintainer, please assign this issue to me. Thanks


ℹ️ **Repo Maintainers**: To accept this application, [review their application](https://www.drips.network/wave/maintainers/issues/6d4c2dcf-6303-46d3-bbc0-c7606a6d956c) or assign @mikevill20 to this issue.

**Comment by Cerome360** [comment]
@Cerome360 has applied to work on this issue as part of the **Stellar Wave Program**'s 7th wave.



> Please assign


ℹ️ **Repo Maintainers**: To accept this application, [review their application](https://www.drips.network/wave/maintainers/issues/6d4c2dcf-6303-46d3-bbc0-c7606a6d956c) or assign @Cerome360 to this issue.
